compose/config/config.py

```diff
@@ -171,6 +171,18 @@ def find_candidates_in_parent_dirs(filenames, path):
     return (candidates, path)
 
 
+def get_services(config, version):
+    if version == 2:
+        return config.get('services', {})
+    else:
+        return config
```

Comment:
Right now this version check is done in 4 places!  This is going to make it much harder to maintain the code. We should check for the config version in only 1 place. This should also result in a smaller change to support the new version.

Since the variation between versions is at the top level of the config, it should be easy enough right now.  Something like this:

1. rename the current `load()` function to `load_services()`
2. create a new `namedtuple('Config', 'service version volumes')`
3. create a new `load()` function that works something like this:

    ```python

    def get_config_version(config_details):
        ... # stubbed out, but returns the version and asserts the version in all files is the same

    def load(config_details):
        version = get_config_version(config_details)
        if not version: # user v1
            return load_services(config_details)
        if version == 2:
            return load_v2(config_details)

        raise UnknownVersionErrror(...)
    ```

4. `load_services()` would return `Config(services, "1", [])`
5. the new `load_v2()` would call `load_services()` to handle the services part, and call another method to handle the volumes validation, and return a `Config` object

This way `load()` always returns the same structure, and the rest of the code doesn't have to know anything about config version.